packages/apollo-links/src/auth-link/clusterAuthLink.ts

```diff
@@ -52,12 +59,27 @@ export class ClusterAuthLink extends ApolloLink {
     });
   }
 
-  private async getUrlAndToken(): Promise<{ url: string; token: string } | undefined> {
-    const nextAgreement = await this.agreementManager.getNextAgreement();
+  private async getRequestParams(): Promise<RequestParams | undefined> {
+    const orderType = this.orderMananger.getNextOrderType();
```

Comment:
You can return ordertype and orderDetail together in one method call.

Reply:
I don't quite understand about it?

Reply:
It seems you've design a workflow, which is step#1: getNextOrderType(), step#2: base on the order type, call getNextAgreement() or getNextPlan()
This implied workflow is not easy to understand, and also not clear what will happen if people don't follow it, for example, even they get the next order type is agreement, they still call getNextPlan(), what's the consequence.